outputs.tf

```
@@ -57,6 +57,11 @@ output "this_rds_cluster_instance_ids" {
   value       = aws_rds_cluster_instance.this.*.id
 }
 
+output "this_rds_cluster_hosted_zone_id" {
+  description = "Route53 hosted zone id of the created cluster"
+  value       = aws_rds_cluster.this[*].hosted_zone_id
```

my apologies @felixb , you were correct before. this is the cluster resource not the cluster instance:
```suggestion
  value       = aws_rds_cluster.this.hosted_zone_id
```